### PR TITLE
Adds writeOut function and restructure

### PR DESCRIPTION
**Pull request by 7oi**
## Premise
I wanted to write junit reports to a file using gulp-lesshint so only logging to console was not optimal. So I looked at the behaviour of other reporters (lesshint-lint-xml-reporter, which borrowed from gulp-jshint-xml-file-reporter) and added that functionality.

## Changes
- Adds writeOut function, which writes out to file if filePath is given. Otherwise to console.
- comment and restructure module
- update version

### CODE REVIEW

**Comment by 7oi** [comment]
@drinkataco Have you managed to check this out? I know it's a bit of a rewrite (I kind of forgot myself in it), but should function the same just with the added functionality of being able to write the report directly to file.

**Comment by drinkataco** [comment]
Hi @7oi, thanks for your contribution.

The writeOut method is definitely needed, so big thanks for adding it.

Just a few issues however, it seems that the test is now failing `yarn test`, and the updates don't follow the style guide `yarn lint`.

I'll be sure to add these tests to the build process, as well as contributing information for the future soon

**Comment by 7oi** [comment]
Ah, so I see. I didn't look further into the Travis output, as it actually seems like it was passing. I also didn't realize there was a style guide defined (I'm quite new to node modules, tbh). I will fix these issues tomorrow and update the PR.

**Comment by drinkataco** [comment]
@7oi I didn't put testing/linting in travis 🤦‍♂️, I'll be sure to fix the build

**Comment by 7oi** [comment]
@drinkataco I have updated the pull request. Here's what happened:
- fixed the tests that were breaking
- the code is now compliant with the linter
- added tests for the writeOut function
- added the linting step to .travis.yml
- added sinon to tests to grab exit code. The previously used method resulted in mocha always exiting with code 0, thus not failing the travis build on failed tests.

**Comment by drinkataco** [comment]
@7oi Sorry I haven't much time to check and merge this PR. I had a quick look tonight, and it seems to me that the tests aren't running correctly (and always exit as an ok) although I haven't been able to delve too much in this. I will prioritise in checking and merging this this week. thanks for your contribution

File: test/test.js [hunk 1]
```diff
@@ -76,28 +78,56 @@ describe('JUNIT Reporter', () => {
...
+    const output = junitReporter.out[0];
+
+    const writeOut = junitReporter.writeOut(filename);
+    writeOut();
+
+    inspect.restore();
+    process.exit.restore();
+
+    // Wait 100 ms before checking if file exists, as writeOut is asynchronous
+    setTimeout(() => {
```
Comment (drinkataco) [inline test/test.js]: Shouldn't we use a promise here if asynchronous, instead of a timeout?

**Review by drinkataco — CHANGES_REQUESTED** [review]

File: test/test.js (hunk 1 above)
Comment (7oi) [reply test/test.js]: Of course. I don't know why I didn't do that originally. I have updated the test.

**Pull request closed without merging**